BridgeAppSDK/SBAUserBridgeManager.m

```diff
@@ -159,9 +159,28 @@ + (void)fetchChangesToScheduledActivities:(NSArray <SBBScheduledActivity *> *)sc
     // Intended design is to allow for the server to win in getting updates to the current list of scheduled
     // activities, but this will also *send* what is already known and may include a finishedOn date that is
     // more recent than what is available from the server. syoung 04/14/2016
-    NSInteger daysAhead = todayOnly ? 0 : 4;
     NSInteger daysBehind = todayOnly ? 0 : 1;
-    [SBBComponent(SBBActivityManager) getScheduledActivitiesForDaysAhead:daysAhead daysBehind:daysBehind cachingPolicy:SBBCachingPolicyFallBackToCached withCompletion:completionBlock];
+    // Always get max daysAhead (4) so we can update local notifications, and then filter to today if necessary.
+    UIApplication *app = [UIApplication sharedApplication];
+    [SBBComponent(SBBActivityManager) getScheduledActivitiesForDaysAhead:4 daysBehind:daysBehind cachingPolicy:SBBCachingPolicyFallBackToCached withCompletion:^(NSArray *activitiesList, NSError *error) {
+        [app cancelAllLocalNotifications];
+        for (SBBScheduledActivity *sa in activitiesList) {
```

Comment:
Shouldn't setting notifications first check that the app has permission? (and wants permission) Since for Parkinsons app the activities are scheduled for midnight... ;)